File: convert_str_to_int/StrToInt.py
def str_to_int(num):
    if len(num) == 0:
        return -1
    pos = 1
    res = 0
    for i in range(len(num) - 1, -1, -1):
        n = int(num[i])
        res += n * pos
        pos *= 10
    return res

if __name__ == "__main__":
    num_str = "356"
    print(str_to_int(num_str))

File: convert_str_to_int/StrToInt.rs
fn str_to_int(num: &str) -> i32 {
    if num.len() == 0 {
        return -1;
    }
    let mut pos = 1;
    let mut res = 0;
    for c in num.chars().rev() {
        let n = c.to_digit(10).unwrap() as i32;
        res += n * pos;
        pos *= 10;
    }
    res
}

fn main() {
    let str = "356";
    println!("{}", str_to_int(str));
}
